### DISCUSSION TITLE
Azure Web App - Current transport is: polling

### DISCUSSION THREAD

**Discussion by huislaw** [miguelgrinberg/Flask-SocketIO#1999] [General]
I managed to deploy the \example on Azure Web App after these steps:
```

1. Installed:
gevent
gevent-websocket

2. Added to requirements.txt:
gevent
gevent-websocket

3. Added .deployment
[config]
SCM_DO_BUILD_DURING_DEPLOYMENT=true
```

However, it is showing **Current transport is: **polling****
In my local machine it is **websocket**

How can I make it to be **websocket** on Azure?

Thanks.

**Comment by miguelgrinberg** [comment]
Did you install a websocket package that is compatible with your chosen async mode? Are you looking at the logs in your server at all? Pretty sure this problem should have been noted there.

**Reply by huislaw** [reply]
1. I installed simple-websocket

2. 
My requirement txt: https://gist.github.com/huislaw/fe13eaeb41c7b656e56112a0e48bb174
My app.py: https://gist.github.com/huislaw/efd060f09bc3ad48ec5d2a46dac06deb

3. Deployed to: https://testhssocketio.azurewebsites.net/
- It shows connected, but Current transport is: polling

4. Deployment log: https://gist.github.com/huislaw/46943d3591157a72275cf8dc997eb86b
- (Doesn't seems to have error)

5. After enabling log for app service I get this error: 
https://gist.github.com/huislaw/c7bc0e3d14c185fc6001ffa86c4685e8
- I'm not sure what is this error about. Do you have any idea?

**Reply by miguelgrinberg** [reply]
Does your deployment platform support websocket?

**Reply by huislaw** [reply]
It is Azure App Service.

It is working now using:
simple-websocket, and startup  "gunicorn -w 1 --threads 1000 app:app" 

Not sure why eventlet and gevent got errors.

Thanks again @miguelgrinberg
